docs/blog/_posts/2018-04-27-eighth-dotty-milestone-release.md

```diff
@@ -0,0 +1,207 @@
+and enables integration with tools such as [Bloop](https://scalacenter.github.io/bloop/).
+
+If you are already using Dotty with sbt 0.13, follow these simple steps to upgrade:
+- update sbt version to 1.1.4 in `project/build.properties`
+- update sbt-dotty plugin to the latest version: `addSbtPlugin("ch.epfl.lamp" % "sbt-dotty" % "0.2.2")`
+- update usages of `.withDottyCompat()` by `.withDottyCompat(scalaVersion.value)`
+
+### Unchecked warnings [#4045](https://github.com/lampepfl/dotty/pull/4045)
+Dotty now emits `unchecked` warnings like `scalac` whenever a type test is performed but cannot
+safely be checked at runtime. For example:
+
+```scala
+scala> def foo(x: Any) = x.isInstanceOf[List[String]]
+1 |def foo(x: Any) = x.isInstanceOf[List[String]]
+  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  |               the type test for List[String] cannot be checked at runtime
+```
+
+In some cases, the Dotty compiler is smarter than `scalac`  and will not emit a warning:
+```scala
+trait Marker
+
+def foo[T](x: T) = x match {
+  case _: T with Marker  => // scalac emits a spurious warning
+  case _ =>
+}
+```
+
+### Kind Polymorphism [#4108](https://github.com/lampepfl/dotty/pull/4108)
+Normally type parameters in Scala are partitioned into kinds. First-level types are types of values.
+Higher-kinded types are type constructors such as `List` or `Map`. The kind of a type is indicated
+by the top type of which it is a subtype. Normal types are subtypes of `Any`, covariant single
+argument type constructors such as List are subtypes of `[+X] => Any`, and the `Map` type
+constructor is a subtype of `[X, +Y] => Any`.
+
+Sometimes we would like to have type parameters that can have more than one kind, for instance to
+define an implicit value that works for parameters of any kind. This is now possible through a form
+of (subtype) kind polymorphism. Kind polymorphism relies on the special type `scala.AnyKind` that
+can be used as an upper bound of a type.
+
+```scala
+def f[T <: AnyKind] = ..
+```
+
+The actual type arguments of f can then be types of arbitrary kinds. So the following would all be
+legal:
+
+```scala
+f[Int]
+f[List]
+f[Map]
+f[[X] => String]
+```
+
+**Note**: This feature is considered experimental and is only enabled under a compiler flag
+(i.e. `-Ykind-polymorphism`). For more information, visit the [Kind Polymorphism](https://dotty.epfl.ch/docs/reference/kind-polymorphism.html)
+section of our documentation.
+
+### Improved support for SAM type [#4152](https://github.com/lampepfl/dotty/pull/4152)
+This release includes fixes to SAM types that greatly improve interoperability with Java 8 lambdas.
+One can now easely write Scala code that uses Java streams:
+
+```scala
+val myList =
+  java.util.Arrays.asList("a1", "a2", "b1", "c2", "c1")
+
+myList
+  .stream
+  .filter(s => s.startsWith("c"))
+  .map(_.toUpperCase)
+  .sorted
+  .forEach(println(_))
+
+// C1
+// C2
```

Comment:
What are these?

Reply:
That's the output of the code above. Maybe we should add `// Prints:`